Add a gendocs pre-submit hook.

brendandburns opened the pull request:
@jlowdermilk @lavalamp

brendandburns committed:
Add a gendocs pre-submit hook.

jbeda commented:
You'll want to update `docs/devel/development.md` too.

Perhaps we should have an "hack/install-git-hooks.sh" helper here?

brendandburns commented:
Updated development.md (and deleted deprecated api doc code).  I don't actually think the script is needed, it's two symlinks, and once you create them, you never have to create them again (even to pick up updates like this).  I can add the script if you really want it.

Thanks!
--brendan

jbeda commented:
Ah -- right.  I had it my mind this would require a new soft link.
